source/d1amp.cpp

```diff
@@ -57,6 +53,25 @@ bool D1Amp::load(QString filePath, int allocate)
     return true;
 }
 
+bool D1Amp::save(SaveAsParam *params)
+{
+    QString selectedPath = params != nullptr ? params->ampFilePath : "";
+    std::optional<QFile> outFile = SaveAsParam::getValidSaveOutput(this->getFilePath(), selectedPath);
+    if (!outFile) {
+        return false;
+    }
+
+    // write to file
+    QDataStream out(*outFile);
```

Comment:
You probably need to first grab the content of the optional and then get a pointer for it:
```suggestion
    QDataStream out(&*outFile);
```
or
```suggestion
    QDataStream out(&outFile->get());
```

Reply:
How do you want to retrieve the selected file-path?

Reply:
Finished revewing things, I'll take a look at this now :)

Reply:
I pushed a fix to the branch, hope you are ok with doing it that way.